lib/active_record_extension.rb

```diff
@@ -0,0 +1,16 @@
+module ActiveRecordExtension
+
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def pluck_with_keys(*keys)
+      self.pluck(*keys).map{ |plucked_values|
+        Hash[keys.zip(plucked_values.kind_of?(Array)? plucked_values : [plucked_values])]
```

Comment:
I would usually do [plucked_values].flatten but no idea if that's good/bad performance-wise...